### PR TITLE
Second phase algorithm #1

### PR DESCRIPTION
**Pull request by BrunoRosendo**

### CODE REVIEW

File: src/bakery/Bakery.cpp [hunk 1]
```diff
@@ -93,32 +97,108 @@ Time Bakery::nearestNeighbour() {
...
+    sort(clients.begin(), clients.end(), [](const Client* c1, const Client* c2) -> bool {
+        return c1->getDeliveryTime() < c2->getDeliveryTime();
+    });
+
+    Vertex *v1 = startingVertex, *v2;
+    Time start(7, 0);
+    for (int i = 0; i < clients.size(); ++i) {
+        Client* client = clients[i];
+        v2 = client->getVertex();
+        graph.dijkstraShortestPath(v1, v2);
```
Comment (monkin77) [inline src/bakery/Bakery.cpp:163]: We could use bidirectional dijkstra when we have fixed vertices. However, we can make that optimization in another pr

**Review by monkin77 — APPROVED** [review]

File: src/bakery/Bakery.cpp (hunk 1 above)
Comment (BrunoRosendo) [reply src/bakery/Bakery.cpp:163]: This should be done but it's a little tricky since I'm using a function to read the vertices' edges (on their path) and the bidirectional dijkstra isn't setting this parameter right, apparently. We need to fix that and add the edges from the last client to the bakery

I wrote TODOs about this too
